apps/zbugs/src/pages/list/list-page.tsx

```diff
@@ -56,54 +56,42 @@ export function ListPage({onReady}: {onReady: () => void}) {
     qs.get('sort')?.toLowerCase() === 'created' ? 'created' : 'modified';
   const sortDirection =
     qs.get('sortDir')?.toLowerCase() === 'asc' ? 'asc' : 'desc';
+  const open =
+    status === 'open' ? true : status === 'closed' ? false : undefined;
+  const pageSize = 10;
+  const [limit, setLimit] = useState(pageSize);
 
-  let q = z.query.issue
+  const q = z.query.issue
     .orderBy(sortField, sortDirection)
     .orderBy('id', sortDirection)
     .related('labels')
-    .related('viewState', q => q.where('userID', z.userID).one());
-
-  const open =
-    status === 'open' ? true : status === 'closed' ? false : undefined;
-
-  if (open !== undefined) {
-    q = q.where('open', open);
-  }
-
-  if (creator) {
-    q = q.whereExists('creator', q => q.where('login', creator));
-  }
-
-  if (assignee) {
-    q = q.whereExists('assignee', q => q.where('login', assignee));
-  }
-
-  if (textFilter) {
-    q = q.where(({or, cmp, exists}) =>
-      or(
-        cmp('title', 'ILIKE', `%${escapeLike(textFilter)}%`),
-        cmp('description', 'ILIKE', `%${escapeLike(textFilter)}%`),
-        exists('comments', q =>
-          q.where('body', 'ILIKE', `%${escapeLike(textFilter)}%`),
-        ),
+    .related('viewState', q => q.where('userID', z.userID).one())
```

Comment:
It's a cool pattern, and this is the best "one-line" version of this that I've seen so far, but honestly I prefer the code as multiple statements.

I like that the query system is composable so that you can construct the query iteratively with normal, simple procedural code.

Reply:
With that I would *almost* be on your side. That's pretty compact.

I think I would still probably write it as multiple statements myself but not certain.